### ISSUE TITLE
Segmentation fault (core dumped) with CPU

### ISSUE THREAD

**Issue by SunHaozhe** [junyanz/pytorch-CycleGAN-and-pix2pix#484]
<img width="695" alt="capture d ecran 2019-01-02 a 19 08 48" src="https://user-images.githubusercontent.com/26926814/50605642-24c7db80-0ec3-11e9-82c8-a0abc6cae7e3.png">

I encountered a problem of segmentation fault (core dumped) when I tried to train a model on horse2zebra using the command:

python train.py --dataroot ./datasets/horse2zebra --name horse2zebra_cyclegan --model cycle_gan --gpu_ids -1

I followed the procedure of README and I am on Google Cloud Platform VM instance with only CPU and Ubuntu 16.04 LTS. I installed python3 using Anaconda3-4.3.0-Linux-x86_64.sh and then installed pytorch using pip install -r requirements.txt. 
<img width="518" alt="capture d ecran 2019-01-02 a 19 23 49" src="https://user-images.githubusercontent.com/26926814/50605870-f72f6200-0ec3-11e9-945e-be5372e55ad0.png">

In fact, it is not only a problem with training on horse2zebra, I have the same problem when I tried to train on other datasets or to test the pretrained models.

How can I solve this problem ?

**Comment by SunHaozhe** [comment]
<img width="1115" alt="1" src="https://user-images.githubusercontent.com/26926814/50606404-ee3f9000-0ec5-11e9-8dd6-a255188295cf.png">
<img width="1115" alt="2" src="https://user-images.githubusercontent.com/26926814/50606411-f4357100-0ec5-11e9-991b-0c4d9326408c.png">
<img width="1120" alt="3" src="https://user-images.githubusercontent.com/26926814/50606414-f992bb80-0ec5-11e9-819b-ae9cdd4df667.png">
<img width="1113" alt="4" src="https://user-images.githubusercontent.com/26926814/50606423-00b9c980-0ec6-11e9-973f-0bbe562f5ef5.png">

This is what I got when I tap:
gdb python
r train.py --dataroot ./datasets/horse2zebra --name horse2zebra_cyclegan --model cycle_gan --gpu_ids -1
where

**Comment by SunHaozhe** [comment]
``

**Comment by junyanz** [comment]
Could you run other PyTorch code with your CPU? Like the code in PyTorch library?

**Comment by junyanz** [comment]
The code works for my CPU, but it is very slow (15 seconds per iteration). Do you have enough CPU memory (~8GB). You can also add `--display_id -1 --print_freq 1` to turn off the visdom display and print training loss every single iteration.

**Comment by rolandakiki** [comment]
I encountered the same problem recently, while testing on 1 CPU. However I still haven't tried increasing my CPU memory on Google cloud platform. Hope this solves the problem.

**Comment by SunHaozhe** [comment]
Thanks for your prompt reply.

Actually it works for me when I train it on an VM instance with GPU. I had a CPU memory less than 8GB and I didn't test the code in PyTorch library. If anyone has a similar problem, he/she could try to increase CPU memory or use a GPU instance. Anyway, it is strange that a problem of type "segmentation fault" could occur in this context.

**Comment by SsnL** [comment]
Please open an issue on [pytorch issue tracker](https://github.com/pytorch/pytorch/issues). Make sure to follow the bug report template.

**Comment by junyanz** [comment]
Cool. You can use a smaller network `--netG resnet_6blocks` with lower image size `--loadSize 143` and `--fineSize 128`.
